### PR TITLE
1544 nginx with domain

### LINKED ISSUE #1544: [SPIKE] Add Nginx configuration - increase security and performance
**Description:**
[Nginx](https://www.nginx.com/) is an HTTP server and reverse proxy. Currently, the backend (Django) is NOT behind a proxy server. This can cause security and/or performance issues for the application. The intention behind this ticket is to add proxy server (Nginx) in front of Gunicorn. 
The proxy server should forward headers, handle static files, rate limit requests, etc.

Note: This is in conjunction with ticket #1543. 

**Acceptance Criteria:** 

- [x] Request header are successfully being forwarded to Gunicorn and Django
- [x] Requests from clients are rate limited
- [x] Static files are cached
- [x] Testing Checklist has run and all tests pass
- [x] README is updated, if necessary

**Tasks:**

- [x] Create config file for Nginx
- [x] Create docker file/add Nginx to docker-compose file
- [x] If ticket #1543 is not implemented yet, then add the command to Gunicorn to trust Nginx
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**

- _The configuration has to be revisited at deployment_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- [UK nginx as firewall sample](https://github.com/alphagov/paas-ip-authentication-route-service)

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

**TODOs:**

- Remove unused CSP vars in settings
- Investigate and remove unused middleware logic
- Must have: Add URLs to backend proxy. Currently only /v1 is added but we need to add any URL schema such as /admin. Also need to add a note to url.py file for future url schemas
- Remove cloud.gov route to the backend.
- Remove "corsheaders" package.
- Must have: add new deployment logic for Nginx, the new network, etc
- Nice to have: Add authentication to Nginx for backend API, so that the URL is not directly reachable without authentication. This need further investigation. One idea is only /auth is reachable and after user is authenticated, then Nginx allows other URLs. This need token authentication between frontend/backend and Nginx

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
This PR includes both Nginx PR (https://github.com/raft-tech/TANF-app/pull/2370) and Domain PR changes (https://github.com/raft-tech/TANF-app/pull/2442).

## How to Test
For testing, this PR has to be deployed staging develop. The deployment should be automatic with changes being pushed to the branch. If staging-develop do not have the changes in this PR (since develop branch is also automatically being pushed), you should trigger deployment in circleci manually.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [x] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [x] what was tested and how
+ [x] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by raftmsohani** [comment]
Had to add network policy:

```
cf add-network-policy tdp-frontend-develop tdp-backend-develop --protocol tcp --port 8080 -o hhs-acf-ofa  -s tanf-staging
```

**Comment by raftmsohani** [comment]
Removed test e2e, since the develop e2e overwrites the base_url to cloud.gov here:
``` 
"test:e2e-develop": "cypress run --headless -b chrome --config baseUrl=https://${BASE_URL_PREFIX}-develop.app.cloud.gov --env cypressToken=${CYPRESS_TOKEN},apiUrl=https://${API_URL_PREFIX}-develop.app.cloud.gov/v1",
```

**Comment by raftmsohani** [comment]
Removed test e2e, since the develop e2e overwrites the base_url to cloud.gov here:
``` 
"test:e2e-develop": "cypress run --headless -b chrome --config baseUrl=https://${BASE_URL_PREFIX}-develop.app.cloud.gov --env cypressToken=${CYPRESS_TOKEN},apiUrl=https://${API_URL_PREFIX}-develop.app.cloud.gov/v1",
```

**Comment by raftmsohani** [comment]
I had sqlite replacing postgres for ease of testing but changed that back to postgres since sqlite had data leaking and migrations not running correctly

File: .circleci/deployment/workflows.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@
           filters:
             branches:
               only:
-                - develop
+                - 1544-Nginx-with-domain # switch back to develop before merge
```
Comment (raftmsohani) [inline .circleci/deployment/workflows.yml]: TODO: To be removed when ready to merge, (only to hijack staging)

File: .circleci/deployment/workflows.yml [hunk 2]
```diff
@@ -37,21 +37,21 @@
...
```
Comment (raftmsohani) [inline .circleci/deployment/workflows.yml]: TODO: To be removed when ready to merge, (only to hijack staging)

File: .circleci/deployment/workflows.yml [hunk 3]
```diff
@@ -37,21 +37,21 @@
...
-                - develop
-      - test-deployment-e2e:
-          requires:
-            - deploy-develop
-          filters:
-            branches:
-              only:
-                - develop
+                - 1544-Nginx-with-domain # switch back to develop before merge
+      #- test-deployment-e2e:
```
Comment (andrew-jameson) [inline .circleci/deployment/workflows.yml]: Need to investigate changing up this line:

tdrs-frontend/package.json
```
    "test:e2e-develop": "cypress run --headless -b chrome --config baseUrl=https://${BASE_URL_PREFIX}-develop.app.cloud.gov --env cypressToken=${CYPRESS_TOKEN},apiUrl=https://${API_URL_PREFIX}-develop.app.cloud.gov/v1",
```

File: .github/workflows/deploy-develop-on-merge.yml [hunk 4]
```diff
@@ -24,6 +24,7 @@ on:
   push:
     branches:
     - develop
+    - 1544-Nginx-with-domain # CHANGE BACK BEFORE MERGE TO DEVELOP
```
Comment (jtimpe) [inline .github/workflows/deploy-develop-on-merge.yml]: todo: remove

File: commands.sh [hunk 5]
```diff
@@ -98,9 +98,16 @@ tdrs-lint-backend() {
     tdrs-compose-backend run --rm web bash -c "flake8 . && if [ $? -eq 0 ]; then echo 'Flake8 linter found no issues'; fi"
 }
 
+# create docker network for tdrs if it doesn't exist
+ tdrs-docker-net() {
+     docker network inspect external-net >/dev/null 2>&1 \
```
Comment (jtimpe) [inline commands.sh:103]: my only concern here is that this _forces_ usage of the frontend `docker-compose.yml` during local development, whereas before we had the option to only run the backend in docker, and run the frontend with `npm run start`. 

the frontend dockerfile performs a development build of the frontend files and serves them via nginx, so we'd lose the hot reload and error reporting functionality of `npm run start` when developing locally. if there's a way around that, i'd like to know because i think it could make future frontend tickets a little more cumbersome to work on.

File: tdrs-frontend/.env.development [hunk 6]
```diff
@@ -3,9 +3,11 @@
 # This file is loaded when running `npm start`
 #
 
-REACT_APP_BACKEND_HOST=http://localhost:8080
+# The hostname behind the tdrs-backend Django app
+REACT_APP_BACKEND_HOST=http://localhost:3000
 
-REACT_APP_BACKEND_URL=http://localhost:8080/v1
+# The URL running the tdrs-backend Django app
+REACT_APP_BACKEND_URL=http://localhost:3000/v1
```
Comment (jtimpe) [inline tdrs-frontend/.env.development:10]: do these values also need to be updated in `cypress.config.js`?

File: tdrs-backend/docker-compose.yml [hunk 7]
```diff
@@ -95,7 +95,6 @@ services:
       bash -c "./wait_for_services.sh &&
                ./gunicorn_start.sh && celery -A tdpservice.settings worker -l info"
     ports:
-      - "8080:8080"
       - "5555:5555"
```
Comment (jtimpe) [inline tdrs-backend/docker-compose.yml:98]: why does the port change? looking through the rest of the configs, i see that gunicorn is still started at `8080`, but i don't see a mapping to that.

File: tdrs-backend/tdpservice/settings/common.py [hunk 8]
```diff
@@ -335,7 +335,7 @@ class Common(Configuration):
     CSP_SCRIPT_SRC = ("'self'", s3_src)
     CSP_IMG_SRC = ("'self'", "data:", s3_src)
     CSP_FONT_SRC = ("'self'", s3_src)
-    CSP_CONNECT_SRC = ("'self'", "*.cloud.gov")
+    CSP_CONNECT_SRC = ("'self'", "*.cloud.gov", "localhost:*")
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/settings/common.py]: can this addition be moved to `settings/local.py`? i don't see it overridden in `settings/cloudgov.py` which seems to imply the CSP would allow connections from `localhost` in deployed/prod environments as well

File: tdrs-frontend/.env. [hunk 9]
```diff
@@ -0,0 +1,50 @@
+###
```
Comment (jtimpe) [inline tdrs-frontend/.env.]: is this file needed?

**Review by jtimpe — CHANGES_REQUESTED** [review]
Testing in the deployed environment and locally looked good! Account request/approval, admin access, data file submission, and submission history all check out.

I have a couple open questions/comments for consideration, but the only changes i'm requesting before approval are the removal of the code blocks marked `todo`.

I'd also like to talk more in the future about the `npm run start` vs `docker-compose up` for running the frontend.

**Review by George-Hudson — APPROVED** [review]
As long as the TODOs get taken care of, this looks good to me

File: commands.sh (hunk 5 above)
Comment (raftmsohani) [reply commands.sh:103]: You can use docker-compose as before, the backend is still available on port 5555, so by only changing the backendURL in frontend env variables, you can build frontend locally and connect it to your backend.

File: tdrs-frontend/.env. (hunk 9 above)
Comment (raftmsohani) [reply tdrs-frontend/.env.]: Answer is NO :)

TODO: TO BE DELETED BEFORE MERGE

File: tdrs-backend/docker-compose.yml (hunk 7 above)
Comment (raftmsohani) [reply tdrs-backend/docker-compose.yml:98]: We do not need the mapping anymore. The external-net takes care of the mapping and can connect to gunicorn using the actual ip+port of instance. That way, we could use the port 8080 on the Nginx in frontend.

File: tdrs-frontend/.env.development (hunk 6 above)
Comment (raftmsohani) [reply tdrs-frontend/.env.development:10]: Have to investigate the cypress config for passing the tests

File: .circleci/deployment/jobs.yml [hunk 10]
```diff
@@ -43,7 +43,7 @@
           app-dir: tdrs-frontend
       - run:
           name: Run Cypress e2e tests
-          command: cd tdrs-frontend; npm run test:e2e-develop
```
Comment (raftmsohani) [inline .circleci/deployment/jobs.yml:46]: TODO: Technical Debt, check having dynamic env var for URLs

**Comment by raftmsohani** [comment]
As per Alex the following IP addresses were provided:
- Guam:
- Virgin Islands:  66.248.166.33 -> It is covered by the following subnet listed in whitelist: 66.248.224.0/19
- PiertoRico: 64.185.209.6 -> It is covered by the following subnet listed in whitelist: 64.185.192.0/19

**Comment by ADPennington** [comment]
> As per Alex the following IP addresses were provided:
> 
> * Guam:
> * Virgin Islands:  66.248.166.33 -> It is covered by the following subnet listed in whitelist: 66.248.224.0/19
> * PiertoRico: 64.185.209.6 -> It is covered by the following subnet listed in whitelist: 64.185.192.0/19

@raftmsohani confirming here that PR was able to successfully access the landing page at https://tdp-frontend-develop.acf.hhs.gov on 3/31/23.

**Comment by raftmsohani** [comment]
Note: Owasp scan was failing because Nginx in the frontend needs the backend to be available, so needed to add the backend docker-compose up as a requirement step to OWASP scan

**Comment by ADPennington** [comment]
> > As per Alex the following IP addresses were provided:
> > 
> > * Guam:
> > * Virgin Islands:  66.248.166.33 -> It is covered by the following subnet listed in whitelist: 66.248.224.0/19
> > * PiertoRico: 64.185.209.6 -> It is covered by the following subnet listed in whitelist: 64.185.192.0/19
> 
> @raftmsohani confirming here that PR was able to successfully access the landing page at https://tdp-frontend-develop.acf.hhs.gov on 3/31/23.

@raftmsohani Guam responded with ip: 67.98.163.2 and confirmed that they can access to the develop frontend landing page.

**Comment by raftmsohani** [comment]
> > > As per Alex the following IP addresses were provided:
> > > 
> > > * Guam:
> > > * Virgin Islands:  66.248.166.33 -> It is covered by the following subnet listed in whitelist: 66.248.224.0/19
> > > * PiertoRico: 64.185.209.6 -> It is covered by the following subnet listed in whitelist: 64.185.192.0/19
> > 
> > 
> > @raftmsohani confirming here that PR was able to successfully access the landing page at https://tdp-frontend-develop.acf.hhs.gov on 3/31/23.
> 
> @raftmsohani Guam responded with ip: 67.98.163.2 and confirmed that they can access to the develop frontend landing page.

Guam is covered by : allow 67.98.168.0/21;

**Comment by ADPennington** [comment]
@raftmsohani can you update the branch and re-deploy?

File: scripts/deploy-frontend.sh [hunk 11]
```diff
@@ -19,29 +19,41 @@ update_frontend()
     cd tdrs-frontend || exit
 
     if [ "$CF_SPACE" = "tanf-prod" ]; then
-        echo "REACT_APP_BACKEND_URL=https://api-tanfdata.acf.hhs.gov/v1" >> .env.production
-        echo "REACT_APP_BACKEND_HOST=https://api-tanfdata.acf.hhs.gov" >> .env.production
+        echo "REACT_APP_BACKEND_URL=https://tanfdata.acf.hhs.gov/v1" >> .env.production
```
Comment (ADPennington) [inline scripts/deploy-frontend.sh:22]: @raftmsohani is this correct? prod backend URL begins with `https://api-tanfdata`

File: scripts/deploy-frontend.sh [hunk 12]
```diff
@@ -19,29 +19,41 @@ update_frontend()
...
+        echo "REACT_APP_FRONTEND_URL=https://tanfdata.acf.hhs.gov" >> .env.production
+        echo "REACT_APP_BACKEND_HOST=https://tanfdata.acf.hhs.gov" >> .env.production
```
Comment (ADPennington) [inline scripts/deploy-frontend.sh:24]: @raftmsohani same q as above ⬆️

**Comment by ADPennington** [comment]
linking to relevant comment: https://github.com/raft-tech/TANF-app/pull/2442#issuecomment-1500500580

login.gov ticket: 3805
ACF Tech prod ticket: REQ0013179
ACF Tech staging ticket: REQ0013178

File: tdrs-frontend/nginx/README.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
+# Nginx Documentation
+
+## 1. Installation
+
+The Nginx uses a master process and mutiple workers to efficiently manage requests. The master process reads and evaluates config file and manages the workers. 
+
+In TDRS, all config and associated files with it are placed in ```/tdrs-frontend/nginx``` folder. There are various differences between the local and deployed versions. Due to the fact that cloud.gov uses buildpacks and local version uses containerized docker instance, there are two separate directories for local and deployed versions.
```
Comment (ADPennington) [inline tdrs-frontend/nginx/README.md]:
```suggestion
In TDP, all config and associated files with it are placed in ```/tdrs-frontend/nginx``` folder. There are various differences between the local and deployed versions. Due to the fact that cloud.gov uses buildpacks and local version uses containerized docker instance, there are two separate directories for local and deployed versions.
```

File: tdrs-frontend/nginx/README.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+## 2. Architecture
+
+In general for TDRS, the *Nginx* handles incoming HTTP requests to both frontend and backend. The frontend requests are handled directly while the backend requests are forwarded to the backend server.
```
Comment (ADPennington) [inline tdrs-frontend/nginx/README.md]:
```suggestion
In general for TDP, the *Nginx* handles incoming HTTP requests to both frontend and backend. The frontend requests are handled directly while the backend requests are forwarded to the backend server.
```

File: tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf
Comment (ADPennington) [inline tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf:1]: it seems like someone could just vpn from one of these whitelisted IPs, since this isnt hidden. i guess this is where access control comes in.

**Review by ADPennington — CHANGES_REQUESTED** [review]
lgtm @raftmsohani 👍🏾  requesting changes to revert temporary syntax needed for testing

File: .circleci/deployment/workflows.yml [hunk 15]
```diff
@@ -37,21 +37,21 @@
...
               only:
-                - develop
-      - test-deployment-e2e:
-          requires:
-            - deploy-develop
-          filters:
-            branches:
-              only:
-                - develop
+                - 1544-Nginx-with-domain # switch back to develop before merge
```
Comment (raftmsohani) [inline .circleci/deployment/workflows.yml]: TODO: To be removed before merge

File: .circleci/deployment/workflows.yml [hunk 16]
```diff
@@ -37,21 +37,21 @@
...
+      #- test-deployment-e2e:
+      #    requires:
+      #      - deploy-develop
+      #    filters:
+      #      branches:
+      #        only:
+      #         - 1544-Nginx-with-domain # switch back to develop before merge
```
Comment (raftmsohani) [inline .circleci/deployment/workflows.yml]: TODO: switch back to develop before merge

File: scripts/deploy-frontend.sh (hunk 11 above)
Comment (raftmsohani) [reply scripts/deploy-frontend.sh:22]: Yes, after merge all urls are served by Nginx which has frontend url

File: tdrs-frontend/.env.development (hunk 6 above)
Comment (raftmsohani) [reply tdrs-frontend/.env.development:10]: Resolved. The cypress test passes and we don't need t change these values

**Comment by ADPennington** [comment]
4/11/update:

login.gov support has confirmed that they made the redirect URI changes. they forgot to follow-up in Feb. should we test before closing the logingov ticket? @raftmsohani

**Comment by ADPennington** [comment]
> 4/11/update:
> 
> login.gov support has confirmed that they made the redirect URI changes. they forgot to follow-up in Feb. should we test before closing the logingov ticket? @raftmsohani

Harold from ACF Tech/OCIO is responsible for the ACF AMS tickets. 4/11/23: We're standing by for an update. @raftmsohani @Smithh-Co

**Comment by ADPennington** [comment]
> > 4/11/update:
> > login.gov support has confirmed that they made the redirect URI changes. they forgot to follow-up in Feb. should we test before closing the logingov ticket? @raftmsohani
> 
> Harold from ACF Tech/OCIO is responsible for the ACF AMS tickets. 4/11/23: We're standing by for an update. @raftmsohani @Smithh-Co

noting for team awareness: Harold made the ACF AMS config changes in staging on 4/11/22. Shared config screenshots offline with @raftmsohani

File: .circleci/deployment/workflows.yml (hunk 1 above)
Comment (George-Hudson) [reply .circleci/deployment/workflows.yml]: @raftmsohani nudge reminder for this one

File: .circleci/deployment/workflows.yml (hunk 3 above)
Comment (George-Hudson) [reply .circleci/deployment/workflows.yml]: @raftmsohani this should be uncommented back I think. It will fail until we re-update the variables in cloud.gov

File: .circleci/deployment/workflows.yml (hunk 2 above)
Comment (George-Hudson) [reply .circleci/deployment/workflows.yml]: reminder to get to this TODO

File: .circleci/deployment/workflows.yml (hunk 15 above)
Comment (George-Hudson) [reply .circleci/deployment/workflows.yml]: reminder :)

File: .circleci/deployment/workflows.yml (hunk 16 above)
Comment (George-Hudson) [reply .circleci/deployment/workflows.yml]: reminder TODO

File: .github/workflows/deploy-develop-on-merge.yml (hunk 4 above)
Comment (George-Hudson) [reply .github/workflows/deploy-develop-on-merge.yml]: reminder TODO

File: tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf
Comment (raftmsohani) [reply tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf:1]: Agree, we should think about AC but the whitelisting through Nginx doesn't replace security settings. This is just another layer

File: scripts/deploy-backend.sh [hunk 17]
```diff
@@ -139,13 +136,16 @@ bind_backend_to_services() {
...
 # provided Cloud.gov App Name
-DEFAULT_ROUTE="https://$CGAPPNAME_BACKEND.app.cloud.gov"
+DEFAULT_ROUTE="https://$CGAPPNAME_FRONTEND.app.cloud.gov"
 if [ -n "$BASE_URL" ]; then
   # Use Shell Parameter Expansion to replace localhost in the URL
   BASE_URL="${BASE_URL//http:\/\/localhost:8080/$DEFAULT_ROUTE}"
 elif [ "$CF_SPACE" = "tanf-prod" ]; then
   # Keep the base url set explicitly for production.
-  BASE_URL="https://api-tanfdata.acf.hhs.gov/v1"
+  BASE_URL="https://tanfdata.acf.hhs.gov/v1"
```
Comment (George-Hudson) [inline scripts/deploy-backend.sh:145]: why does this change? Just saw this.

Comment (George-Hudson) [reply scripts/deploy-backend.sh:145]: NM you answered below :)

File: .circleci/build-and-test/jobs.yml [hunk 18]
```diff
@@ -42,7 +42,7 @@
           path: tdrs-frontend/pa11y-screenshots/
 
   test-e2e:
-    executor: machine-executor
+    executor: zap-machine
```
Comment (George-Hudson) [inline .circleci/build-and-test/jobs.yml]: make sure you merge my branch again, or develop, now that my branch merged into develop with a minor change to make this show `large-machine-executor`

File: tdrs-backend/tdpservice/settings/cloudgov.py [hunk 19]
```diff
@@ -158,7 +158,10 @@ class Staging(CloudGov):
     """Settings for applications deployed in the Cloud.gov staging space."""
 
     # TODO: why not just 'appcloudgov'?
```
Comment (George-Hudson) [inline tdrs-backend/tdpservice/settings/cloudgov.py]: Can probably get rid of this TODO since we are specifically not using the `app.cloud.gov` in the TANF-Staging cf space anymore.

File: tdrs-frontend/.env. (hunk 9 above)
Comment (George-Hudson) [reply tdrs-frontend/.env.]: Reminder to delete :)

File: tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf
Comment (George-Hudson) [reply tdrs-frontend/nginx/cloud.gov/ip_whitelist.conf:1]: Maybe discuss this some more? Or create a spike to see if we can further increase security around this?

File: tdrs-frontend/package.json [hunk 20]
```diff
@@ -39,7 +39,6 @@
     "test:accessibility": "concurrently -k -s first 'REACT_APP_PA11Y_TEST=true npm run start:ci' 'wait-on http://localhost:3000/ && npm run pa11y-ci'",
     "test:e2e": "cypress open",
     "test:e2e-ci": "cypress run --headless -b chrome --env cypressToken=${CYPRESS_TOKEN}",
-    "test:e2e-develop": "cypress run --headless -b chrome --config baseUrl=https://${BASE_URL_PREFIX}-develop.app.cloud.gov --env cypressToken=${CYPRESS_TOKEN},apiUrl=https://${API_URL_PREFIX}-develop.app.cloud.gov/v1",
```
Comment (George-Hudson) [inline tdrs-frontend/package.json:42]: This was meant to be a full integration test for the develop environment with all actual pieces (front-end, back-end, and all attached services) interacting, not testing each piece individually. Is there a replacement put in place I am not seeing?

Comment (raftmsohani) [reply tdrs-frontend/package.json:42]: Yes, we will need to use test:e2e-ci, and use inline options to override urls, etc

**Comment by raftmsohani** [comment]
> 4/11/update:
> 
> login.gov support has confirmed that they made the redirect URI changes. they forgot to follow-up in Feb. should we test before closing the logingov ticket? @raftmsohani

Yes, I think it's a good idea to test.

**Comment by raftmsohani** [comment]
> > 4/11/update:
> > login.gov support has confirmed that they made the redirect URI changes. they forgot to follow-up in Feb. should we test before closing the logingov ticket? @raftmsohani
> 
> Yes, I think it's a good idea to test.

@andrew-jameson

**Comment by andrew-jameson** [comment]
Per Dev sync 4/12, discussed merging to `develop` today and testing sandbox on the staging space. We will test production on Friday 4/14 using the prod login.gov setup and changes to ACF AMS will also be done at that time.

**Review by ADPennington — APPROVED** [review]

**Review by jtimpe — APPROVED** [review]
